src/logic/hosts.rs

```diff
@@ -17,53 +17,63 @@ pub enum HostError {
     AlreadyLeased(Vec<HostId>),
 }
 
-pub async fn get_available_hosts(registry: &Registry) -> Result<Vec<Host>, HostError> {
-    let mut tx = registry.begin().await?;
-    let hosts = tx.get_available_hosts().await?;
-    tx.commit().await?;
-    Ok(hosts)
+pub struct HostsStory {
```

Comment:
HostsService ?

Reply:
Is there an advantage to using the struct with methods instead of standalone functions?